ow_regolith/include/ServiceClientFacade.h

```diff
@@ -0,0 +1,84 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+#ifndef SERVICE_CLIENT_FACADE_H
+#define SERVICE_CLIENT_FACADE_H
+
+#include <memory>
+
+#include <ros/ros.h>
+
+namespace ow_regolith {
+
+template <class T>
+class ServiceClientFacade
+{
+public:
+  ServiceClientFacade() : m_node_handle(nullptr), m_persistent(false)
+  {
+    // do nothing
+  }
+  ServiceClientFacade(const ServiceClientFacade&) = delete;
+  ~ServiceClientFacade() = default;
+  ServiceClientFacade& operator=(const ServiceClientFacade&) = delete;
+
+  bool connect(std::shared_ptr<ros::NodeHandle> &nh,
+               const std::string &service_path,
+               ros::Duration timeout, bool persistent)
+  {
+    m_node_handle = nh;
+    m_persistent = persistent;
+    m_client = m_node_handle->serviceClient<T>(service_path, m_persistent);
+    if (!m_client.waitForExistence(timeout)) {
+      ROS_ERROR("Timed out waiting for service %s to advertise",
+                service_path.c_str());
+      return false;
+    }
+    return true;
+  }
+
+  bool call(T &message)
+  {
+    if (!m_node_handle) {
+      ROS_ERROR(
+        "ServiceClientFacade called before being connected."
+      );
+      return false;
+    }
+
+    if (!m_client.isValid() && !attempt_reconnect()) {
+      ROS_ERROR(
+        "Connection to service %s has been lost and reconnection failed",
+        m_client.getService().c_str()
+      );
+      return false;
+    }
+
+    if (!m_client.call(message)) {
+      ROS_ERROR("Failed to call service %s", m_client.getService().c_str());
+      return false;
+    }
+    return true;
+  }
+
```

Comment:
The `connect` and `call` functions are rather long for inline functions.  The RegolithSpawner has just one Service but if there were many services (in this or another user), there would be some code bloat.   Suggest either un-inlining or explaining the rationale for inlining in a comment.

Reply:
Inlining was unnecessary. They are now implemented in the same header file below the class definition. The original reason for them being implemented in the same header file was because it was a template class; however, in our conversation this morning we both realized `ServiceClientFacade` need not be a template class at all, only its member functions need to be. This is now the case. However, I decided to keep member function implementation inside the header file because this makes the class more portable to other software packages, and you expressed interest in using it in **ow_autonomy**.

Making `ServiceClientFacade` a non-template class has also made the class more useful since the message type no longer needs to be decided upon its instantiation, but instead can be decided or even changed for a single instance each time the connect method is called. This allows a single `ServiceClientFacade` to handle connections to multiple different ROS Services, so long as they are not concurrent connections.